### PR TITLE
Wordwrap fix: #721

### PR DESCRIPTION
**Pull request by codingtwinky**
screen shot 2016-03-31 at 7 19 03 pm

### CODE REVIEW

**Comment by FredrikNoren** [comment]
You're a monster man :) good stuff. Two things: Should have a clicktest for this (it will break in the future otherwise) and add to CHANGELOG

File: components/commit/commit.html [hunk 1]
```diff
@@ -28,6 +28,14 @@
         <div class="body" data-bind="text: body"></div>
         <div class="diff-wrapper" data-bind="visible: showCommitDiff, style: diffStyle, click: stopClickPropagation">
           <div class="btn-group btn-group-xs" data-bind="visible: selected">
+            <button class="btn btn-default bootstrap-tooltip pull-right" data-ta-clickable="commit-worldwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Word Wrapp'"></span>
```
Comment (campersau) [inline components/commit/commit.html]: `'Word Wrap'`?

File: components/commit/commit.js [hunk 2]
```diff
@@ -67,7 +68,9 @@ CommitViewModel.prototype.setData = function(args) {
       sha1: this.sha1,
       repoPath: this.repoPath,
       server: this.server,
-      textDiffType: this.textDiffType }));
+      textDiffType: this.textDiffType,
+      wordWrap: this.wordWrap }));
+      console.log(44, this);
```
Comment (campersau) [inline components/commit/commit.js]: Remove `console.log`

File: components/staging/staging.html [hunk 3]
```diff
@@ -48,12 +48,22 @@
...
+              <span data-bind="text: 'Side by Side'"></span>
+            </button>
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-defaultDiff" data-bind="click: textDiffTypeChange.bind($data, 'textdiff'), css: {active: textDiffType() === 'textdiff'}" data-toggle="tooltip" data-placement="bottom" data-original-title="Show inline diff view" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Default'"></span>
+            </button>
+          </div>
+          <div class="btn-group pull-right btn-group-sm">
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="commit-worldwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Word Wrapp'"></span>
```
Comment (campersau) [inline components/staging/staging.html]: `'Word Wrap'`?

File: components/staging/staging.html [hunk 4]
```diff
@@ -48,12 +48,22 @@
...
+              <span data-bind="text: 'Side by Side'"></span>
+            </button>
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-defaultDiff" data-bind="click: textDiffTypeChange.bind($data, 'textdiff'), css: {active: textDiffType() === 'textdiff'}" data-toggle="tooltip" data-placement="bottom" data-original-title="Show inline diff view" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Default'"></span>
+            </button>
+          </div>
+          <div class="btn-group pull-right btn-group-sm">
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="commit-worldwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
```
Comment (campersau) [inline components/staging/staging.html]: `commit-worldwrap` --> `commit-wordwrap`

File: components/commit/commit.html [hunk 5]
```diff
@@ -28,6 +28,14 @@
         <div class="body" data-bind="text: body"></div>
         <div class="diff-wrapper" data-bind="visible: showCommitDiff, style: diffStyle, click: stopClickPropagation">
           <div class="btn-group btn-group-xs" data-bind="visible: selected">
+            <button class="btn btn-default bootstrap-tooltip pull-right" data-ta-clickable="commit-worldwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
```
Comment (campersau) [inline components/commit/commit.html]: `commit-worldwrap` --> `commit-wordwrap`

File: components/textdiff/textdiff.js [hunk 6]
```diff
@@ -22,6 +22,8 @@ var TextDiffViewModel = function(args) {
   this.isShowingDiffs = args.isShowingDiffs;
   this.diffProgressBar = args.diffProgressBar;
   this.editState = args.editState;
+  this.wordWrap = args.wordWrap;
+  console.log(111, this, args);
```
Comment (campersau) [inline components/textdiff/textdiff.js]: Remove `console.log`

File: components/commit/commit.js [hunk 7]
```diff
@@ -86,3 +89,6 @@ CommitViewModel.prototype.textDiffTypeChange = function(type) {
 CommitViewModel.prototype.stopClickPropagation = function(data, event) {
   event.stopImmediatePropagation();
 }
+CommitViewModel.prototype.toggleWordWrap = function(state) {
+  this.wordWrap(state);
```
Comment (campersau) [inline components/commit/commit.js]: I would either change the function to do this `this.wordWrap(!this.wordWrap());` or rename it to `wordWrapChange` (like `textDiffTypeChange`)

File: components/staging/staging.js [hunk 8]
```diff
@@ -292,8 +293,11 @@ StagingViewModel.prototype.onAltEnter = function(d, e){
     }
     return true;
 };
+StagingViewModel.prototype.toggleWordWrap = function(state) {
+  this.wordWrap(state);
```
Comment (campersau) [inline components/staging/staging.js]: I would either change the function to do this `this.wordWrap(!this.wordWrap());` or rename it to `wordWrapChange` (like `textDiffTypeChange`)

Comment (codingtwinky) [inline components/staging/staging.js]: So the problem with that is that we have two buttons, "default" and "wrap". if we do `this.wordWrap(!this.wordWrap());` then when a person click "wrap" button twice its gets weird... 

I may just change this button and side by side view buttons to a toggle button later...

File: clicktests/test.generic.js [hunk 9]
```diff
@@ -95,9 +95,19 @@ suite.test('Test showing commit side by side diff between two commits', function
   helpers.click(page, '[data-ta-clickable="commit-sideBySideDiff"]');
   helpers.waitForElementVisible(page, '[data-ta-container="commitLineDiffs"]', function() {
     setTimeout(function() {                           // let it finish making api call
-      helpers.click(page, '[data-ta-clickable="node-clickable-0"]'); // De-select again
+      // helpers.click(page, '[data-ta-clickable="node-clickable-0"]'); // De-select again
```
Comment (campersau) [inline clicktests/test.generic.js]: Why did you change this?

Comment (codingtwinky) [inline clicktests/test.generic.js]: This is because we need to keep "node clicked" state.  if we don't keep the state test will fail as word wrap button will not be visible.
